### PR TITLE
DNS utils improvements

### PR DESCRIPTION
**Pull request by The-Loeki**
### What does this PR do?
Fix various issues with the dns utils, expand it's functionality and increase it's testing coverage

### New Behavior
* Add TLSA, CAA and SSHFP 'intelligent' records 
* Various bugs and corner cases fixed
* remove duplicate code and functions 
* add solid TLD determination when available

### Previous PR's
#39639
#40269

### Tests written?
Yes

### CODE REVIEW

**Comment by garethgreenaway** [comment]
@The-Loeki Looks like some of the DNS tests are faililng.

**Comment by The-Loeki** [comment]
@garethgreenaway / @terminalmage 
I'd like to ask you guys to merge this if that's OK; I've got the equivalent replacement module WIP as well pending that.

I'd like your opinion before continuing as well:
I've noticed for 2018.3 the `refresh_dns()` and `dns_check()` functions got moved to `salt.utils.network`
That IMHO isn't that logical considering `salt.utils.dns.etcetera()`

So:
* could you care less?
* or make `salt.utils.network.dns` and refactor the lot into that
* or refactor `dns_check()` & `refresh_dns()` to `salt.utils.dns.check()|refresh_resolv(()`

**Comment by terminalmage** [comment]
I'm not familiar enough with this DNS utils module to comment on it at the moment.

**Comment by garethgreenaway** [comment]
@The-Loeki Probably makes more sense and more consistent to move all dns related utils into salt.utils.dns

**Review by garethgreenaway — APPROVED** [review]

File: salt/utils/dns.py [hunk 1]
```diff
@@ -887,6 +966,21 @@ def sshfp_data(key_t, hash_t, pub):
     return _rec2data(key_t, hash_t, ssh_fp)
 
 
+def sshfp_rec(rdata):
+    '''
+    Validate and parse DNS record data for TLSA record(s)
```
Comment (skorpy2009) [inline salt/utils/dns.py]: shouldn't this be `sshfp record(s)`

**Pull request merged**